k3k-kubelet/provider/token.go

```diff
@@ -121,6 +121,15 @@ func removeKubeAccessVolume(pod *corev1.Pod) {
 		}
 	}
 
+	// ephemeral containers
+	for i, container := range pod.Spec.EphemeralContainers {
+		for j, mountPath := range container.VolumeMounts {
+			if strings.HasPrefix(mountPath.Name, kubeAPIAccessPrefix) {
+				pod.Spec.EphemeralContainers[i].VolumeMounts = append(pod.Spec.EphemeralContainers[i].VolumeMounts[:j], pod.Spec.EphemeralContainers[i].VolumeMounts[j+1:]...)
+			}
+		}
+	}
```

Comment:
This append, and the previous/next one, could not work, because we are appending while iterating on the same array.

https://go.dev/play/p/WdtghMPTTUn

Do we have multiple of these api access tokens? If not we could `break` the loop.